examples/nextjs-spa/pages/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import Link from "next/link"
+
+// React
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+// Ory SDK
+import { ory } from "../components/sdk"
+
+// import css
+import styles from "../styles/Dashboard.module.css"
+
+import { AxiosError } from "axios"
+import type { NextPage } from "next"
+import { LogoutLink } from "../pkg/hooks"
+
+const Home: NextPage = () => {
+  const [session, setSession] = useState<string>(
+    "No valid Ory Session was found.\nPlease sign in to receive one.",
+  )
+  const [hasSession, setHasSession] = useState<boolean>(false)
+  const router = useRouter()
+  const onLogout = LogoutLink()
+
+  useEffect(() => {
+    ory
+      .toSession()
+      .then(({ data }) => {
+        setSession(session)
+        setHasSession(true)
+      })
+      .catch((err: AxiosError) => {
+        switch (err.response?.status) {
+          case 403:
+          // This is a legacy error code thrown. See code 422 for
+          // more details.
+          case 422:
+            // This status code is returned when we are trying to
+            // validate a session which has not yet completed
+            // it's second factor
+            return router.push("/login?aal=aal2")
```

Comment:
handle this error correctly. sometimes it is not just 2fa

you can check how we handle errors like these 
https://github.com/ory/kratos-selfservice-ui-react-nextjs/blob/master/pkg/errors.tsx